dali/daliadmin/daadmin.cpp

```diff
@@ -3461,4 +3462,66 @@ void removeOrphanedGlobalVariables(bool dryrun, bool reconstruct)
     }
 }
 
+
+void cleanStaleGroups(const char *groupPattern, bool dryRun)
+{
+    PROGLOG("Collecting group names from logical files");
+    Owned<IRemoteConnection> conn = querySDS().connect("/Files", myProcessSession(), 0, daliConnectTimeoutMs);
+
+    std::unordered_set<std::string> fileGroups;
+    Owned<IPropertyTreeIterator> filesIter = conn->queryRoot()->getElements("//File", iptiter_remote);
+    ForEach(*filesIter)
+    {
+        IPropertyTree &file = filesIter->query();
+        const char *group = file.queryProp("@group");
+        if (!isEmptyString(group))
+            fileGroups.insert(group);
+    }
+    PROGLOG("Found %zu unique groups in /Files", fileGroups.size());
+    conn.clear();
+
+    PROGLOG("Collecting group names from /Groups");
+    conn.setown(querySDS().connect("/Groups", myProcessSession(), dryRun ? 0 : RTM_LOCK_WRITE, daliConnectTimeoutMs));
+    if (!conn)
+    {
+        PROGLOG("Failed to connect to /Groups");
+        return;
+    }
+
+    std::unordered_set<std::string> groupToDelete;
+
+    StringBuffer xpath("Group");
+    if (!isEmptyString(groupPattern))
+        xpath.appendf("[@name='%s']", groupPattern);
```

Comment:
Should this be ~ rather than =?

Reply:
yes, =~ , I'll change, though in practice it makes no difference, the match is always interpreted as wild with or without ~, the idea was that = would only mean hard match, but was never changed in fear of breaking something.